code/modules/power/singularity/singularity.dm

```diff
@@ -30,9 +30,11 @@ var/list/global_singularity_pool
 	var/last_warning
 	appearance_flags = LONG_GLIDE
 	var/chained = 0 //Adminbus chain-grab
+	var/modifier = "" //for memes
```

Comment:
Does the singularity even have the ability to update icons? If so you should slap the modifier somewhere in icon state updating or something so that it can actually kick in the moment any admin changes the modifier

Reply:
I'm not going to completely fix the epic shitcode that is the singularity switch statement but I'll move the icon check to its process so it updates dynamically